Build/scripts/init_build.ps1

```diff
@@ -76,39 +76,35 @@ if ($YearAndMonth) {
 }
 
 $BuildIdentifier = "${PushID}_${PushDate}_${Username}_${CommitHash}"
-$OutputPath = "${BranchPath}\${YearAndMonth}${YearAndMonthSeparator}${BuildIdentifier}"
-$FlavorName = "${Env:BuildPlatform}_${Env:BuildConfiguration}"
+$ComputedDropPathSegment = "${BranchPath}\${YearAndMonth}${YearAndMonthSeparator}${BuildIdentifier}"
+$BuildType = "${Env:BuildPlatform}_${Env:BuildConfiguration}"
+$BinariesDirectory = "${Env:BUILD_SOURCESDIRECTORY}\Build\VcBuild"
 
-# Create a sentinel file for each build flavor and for the overall build
-# to track whether the build is complete.
-# * build.incomplete                   # will be deleted when the release task completes
+# Create a sentinel file for each build flavor to track whether the build is complete.
 # * ${arch}_${flavor}.incomplete       # will be deleted when the build of this flavor completes
 
 $buildIncompleteFileContentsString = @"
 {0} in progress.
 The contents of this directory should not be relied on until the build completes.
 "@
 
-$FullOutputPath = Join-Path ${env:DROP_ROOT} ${OutputPath}
-New-Item -ItemType Directory -Force -Path $FullOutputPath
-New-Item -ItemType Directory -Force -Path (Join-Path $Env:BUILD_BINARIESDIRECTORY "buildlogs")
-New-Item -ItemType Directory -Force -Path (Join-Path $Env:BUILD_BINARIESDIRECTORY "logs")
+$DropPath = Join-Path $env:DROP_ROOT $ComputedDropPathSegment
+New-Item -ItemType Directory -Force -Path $DropPath
+New-Item -ItemType Directory -Force -Path (Join-Path $Env:BUILD_SOURCESDIRECTORY "test\logs")
+New-Item -ItemType Directory -Force -Path (Join-Path $BinariesDirectory "testlogs")
+New-Item -ItemType Directory -Force -Path (Join-Path $BinariesDirectory "buildlogs")
+New-Item -ItemType Directory -Force -Path (Join-Path $BinariesDirectory "logs")
 
-$buildIncompleteFile = Join-Path $FullOutputPath "build.incomplete"
-$flavorBuildIncompleteFile = Join-Path $FullOutputPath "${FlavorName}.incomplete"
+$flavorBuildIncompleteFile = Join-Path $DropPath "${BuildType}.incomplete"
 
-if (-not (Test-Path $buildIncompleteFile)) {
-    ($buildIncompleteFileContentsString -f "Build") `
-        | Out-File $buildIncompleteFile -Encoding Ascii
-}
 if (-not (Test-Path $flavorBuildIncompleteFile)) {
-    ($buildIncompleteFileContentsString -f "Build of ${FlavorName}") `
+    ($buildIncompleteFileContentsString -f "Build of ${BuildType}") `
         | Out-File $flavorBuildIncompleteFile -Encoding Ascii
 }
 
 $PogoConfig = "False"
-if (((${Env:BuildPlatform} -eq "x64") -or (${Env:BuildPlatform} -eq "x86")) `
-    -and (${Env:BuildConfiguration} -eq "release"))
+if ((($Env:BuildPlatform -eq "x64") -or ($Env:BuildPlatform -eq "x86")) `
```

Comment:
Excellent idea. I've decided to make those arrays into script parameters as well so that we can customize them from the build definition if need be.

---
In reply to: [63264916](https://github.com/Microsoft/ChakraCore/pull/981#discussion_r63264916) [](ancestors = 63264916)